DigitalLearningSolutions.Data/Services/UserService.cs

```diff
@@ -193,77 +201,75 @@ List<DelegateUser> delegateUsers
         }
 
         // TODO HEEDLS-887 Make sure this logic is correct with the new account structure
-        public void UpdateUserAccountDetailsForAllVerifiedUsers(
+        public void UpdateUserDetailsAndCentreSpecificDetails(
+            int userId,
             MyAccountDetailsData myAccountDetailsData,
-            CentreAnswersData? centreAnswersData = null
+            CentreAnswersData centreAnswersData

```

Comment:
Split this into two params
* `string? centreEmail`
* `DelegateDetailsData? delegateDetailsData`

Such that if delegateDetailsData is non-null then the user is assumed to have a delegate account. 
Just noticed delegateId is on MyAccountDetailsData - might make sense to port that across to delegateDetailsData.